### PR TITLE
Add integration test into travis

### PR DESCRIPTION
**Pull request by nojunpark**

### CODE REVIEW

**Comment by nojunpark** [comment]
@JohanLorenzo everything passed, but the checks have failed with 1, probably something to do with the kill command, how can I fix this?

**Comment by nojunpark** [comment]
oh actually, that's not it, hmm i wonder why

**Comment by nojunpark** [comment]
@JohanLorenzo r?

**Comment by JohanLorenzo** [comment]
~~R+ awesome!~~

Edit: Removed r+. See [this comment](https://github.com/fxbox/foxbox/pull/205#issuecomment-196763898)

**Comment by JohanLorenzo** [comment]
Thanks for the investigation No-Jun. Between [this comment](https://github.com/fxbox/foxbox/pull/205#issuecomment-196428869) and your rebase, [that line landed](https://github.com/fxbox/foxbox/pull/183/files#diff-6f414de556ea07ebc3f46c66dab412ecR89). 

No changes were made in .travis.yml there. Hence, since then, CI hasn't clean up itself correctly. Let's use this PR as a follow up of #183 to make sure we're in a clean state.

File: .travis.yml [hunk 1]
```diff
@@ -29,8 +29,10 @@ before_script:
     - npm install
 
 script:
-    - rm -f *sqlite
+    #- rm -f *sqlite
```
Comment (JohanLorenzo) [inline .travis.yml]: Nit: Remove commented code

File: .travis.yml [hunk 2]
```diff
@@ -29,8 +29,10 @@ before_script:
...
     - cargo build # Linter is also executed here
     - $TRAVIS_BUILD_DIR/tools/execute-unit-tests-with-coverage
-    - jshint static/main/js/*.js static/setup/js/*.js test/selenium/*.js
-    - (cargo run -- -l $BOX_LOCAL_NAME -p $BOX_PORT &> /dev/null &) ; npm test
+    - jshint static/main/js/*.js static/setup/js/*.js test/selenium/*.js test/integration/lib/*.js test/integration/test/*.js
+    - npm run test-integration # starts foxbox and kills it within the test script
+    - rm /home/travis/.local/share/foxbox/users_db.sqlite
```
Comment (JohanLorenzo) [inline .travis.yml]: As the path is not that trivial anymore, I'd the removal process in a script (a npm one should be fine) and call that script here.

For instance:

``` sh
npm run clean
```

File: package.json [hunk 3]
```diff
@@ -6,7 +6,9 @@
     "node": ">=4.0.0"
   },
   "scripts": {
-    "test": "rm -f *sqlite && ./node_modules/.bin/mocha test/selenium/*_test.js"
+    "test-selenium": "rm -f *sqlite && ./node_modules/.bin/mocha test/selenium/*_test.js",
```
Comment (JohanLorenzo) [inline package.json]: `rm -f *sqlite` doesn't do anything anymore. I'd run the clean script instead

File: package.json [hunk 4]
```diff
@@ -6,7 +6,9 @@
...
+    "test-integration": "./node_modules/.bin/mocha test/integration/test/*.js",
+    "test": "npm test-selenium && npm test-integration"
```
Comment (JohanLorenzo) [inline package.json]: Nit: `run` is missing in both commands

File: src/controller.rs [hunk 5]
```diff
@@ -96,7 +96,7 @@ impl Controller for FoxBox {
 
     fn run(&mut self, shutdown_flag: &AtomicBool) {
         debug!("Starting controller");
-
+        info!("sqlite location: {}", self.profile_service.path_for("users_db.sqlite"));
```
Comment (JohanLorenzo) [inline src/controller.rs]: That's a hardcoded string. I'd either remove this line or extract the path retrieval in a private function.

File: test/integration/test/philips_light_test.js [hunk 6]
```diff
@@ -7,8 +7,9 @@ const fs = require('fs');
 
 var philipshue_server = require('../lib/philipsHue_server.js');
 var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
-var config = new Config('./lib/config/foxbox.js');
-var header = new Config('./lib/config/header.js').get('header');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var header = new Config('./test/integration/lib/config/header.js')
+.get('header');
```
Comment (JohanLorenzo) [inline test/integration/test/philips_light_test.js]: Nit: I missed it in my preview review, but having `header.header` looks redundant. How about changing header.js with something like:

``` js
module.exports = {
  'Content-Type' : 'application/json'
};
```

?

File: test/integration/test/philips_light_test.js [hunk 7]
```diff
@@ -29,7 +30,7 @@ describe('Initiate the connection with foxbox as Philips Hue hub',function(){
     philipshue_server.turnOffLight(2);
     philipshue_server.turnOffLight(3);
 
-    // kick off the foxbox within 10 seconds
+    // kick off the foxbox within 3 seconds
```
Comment (JohanLorenzo) [inline test/integration/test/philips_light_test.js]: I don't see this value coded somewhere. Is is an approximation of the time taken, or is it a hard deadline?
I'd clarify the comment.

**Comment by JohanLorenzo** [comment]
~~r? @JohanLorenzo~~ 
Due to the changes needed for the clean up, we have to perform some small extra-changes. In order to prevent history to repeat itself, please ask me for review once they are made and commits are squashed :smiley:

**Comment by JohanLorenzo** [comment]
r+

LGTM!

**Pull request merged**